e2e/e2e_test.go

```diff
@@ -0,0 +1,301 @@
+			}
+			l.Printf("Took %v for nodes to become ready\n", time.Since(start))
+
+			// Check that load balancer is working.
+
+			// Install example pod to load-balance to.
+			appName := "app"
+			pod := corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: appName,
+					Labels: map[string]string{
+						"app": appName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "nginx",
+							Image: "nginx",
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 80,
+								},
+							},
+						},
+					},
+				},
+			}
+
+			if _, err := cs.CoreV1().Pods(corev1.NamespaceDefault).Create(&pod); err != nil {
+				t.Fatalf("failed to create example pod: %s", err)
+			}
+
+			// Wait for example pod to become ready.
+			l.Println("Polling for pod readiness")
+			start = time.Now()
+			var appPod *corev1.Pod
+			if err := wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
+				pod, err := cs.CoreV1().Pods(corev1.NamespaceDefault).Get(appName, metav1.GetOptions{})
+				if err != nil {
+					if kerrors.IsNotFound(err) {
+						return false, nil
+					}
+					return false, err
+				}
+				appPod = pod
+				for _, cond := range appPod.Status.Conditions {
+					if cond.Type == corev1.PodReady && cond.Status == corev1.ConditionTrue {
+						return true, nil
+					}
+				}
+				return false, nil
+			}); err != nil {
+				t.Fatalf("failed to observe ready example pod %q in time: %s\npod: %v", appName, err, appPod)
+			}
+			l.Printf("Took %v for pod to become ready\n", time.Since(start))
+
+			// Create service object.
+			svcName := "svc"
+			svc := corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: svcName,
+				},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{
+						"app": appName,
+					},
+					Type: corev1.ServiceTypeLoadBalancer,
+					Ports: []corev1.ServicePort{
+						{
+							Port: 80,
+						},
+					},
+				},
+			}
+
+			if _, err := cs.CoreV1().Services(corev1.NamespaceDefault).Create(&svc); err != nil {
+				t.Fatalf("failed to create service: %s", err)
+			}
+			// External LBs don't seem to get deleted when the kops cluster is
+			// removed, at least not on DO. Hence, we'll do it explicitly.
+			defer func() {
+				if err := cs.CoreV1().Services(corev1.NamespaceDefault).Delete(svcName, &metav1.DeleteOptions{}); err != nil {
+					t.Fatalf("failed to delete service: %s", err)
+				}
+			}()
+
+			// Wait for service IP address to be assigned.
+			l.Println("Polling for service load balancer IP address assignment")
+			start = time.Now()
+			var lbAddr string
+			if err := wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
+				svc, err := cs.CoreV1().Services(corev1.NamespaceDefault).Get(svcName, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				for _, ing := range svc.Status.LoadBalancer.Ingress {
```

Comment:
I think this for loop here is a bit confusing cause we return on first index. Should just check for length and use `svc.Status.LoadBalancer.Ingress[0].IP`.

Reply:
Sure, makes sense -- done!